app/src/main/java/com/github/se/wanderpals/model/viewmodel/NotificationsViewModel.kt

```diff
@@ -114,20 +123,32 @@ open class NotificationsViewModel(val tripsRepository: TripsRepository, val trip
    * @return A boolean representing the success of the operation.
    */
   open fun addAnnouncement(announcement: Announcement) {
-    runBlocking {
-      tripsRepository.addAnnouncementToTrip(tripId, announcement)
-      val listOfTokens = tripsRepository.getTrip(tripId)
-      if (listOfTokens != null) {
-        Log.d("NotificationAnnouncement", "List of tokens: ${listOfTokens.tokenIds}")
-      }
-      if (listOfTokens != null) {
-        for (token in listOfTokens.tokenIds) {
-          sendMessageToListOfUsers(token, announcement.title)
+    viewModelScope.launch {
+      if (!isAddingAnnouncement.value && !createAnnouncementFinished.value) {
+
+        _isAddingAnnouncement.value = true
+        tripsRepository.addAnnouncementToTrip(tripId, announcement)
+        val listOfTokens = tripsRepository.getTrip(tripId)
+        if (listOfTokens != null) {
```

Comment:
You can safely delete this if condition, let only the second one. It will decrease the complexity of the code

Reply:
Done.)